### PR TITLE
[task-submit] Submit task for review

### PR DESCRIPTION
**Pull request by jgerigmeyer**
See: https://trello.com/c/s71Jlbye

butterfly_puppy

## Steps to reproduce

First, run `python manage.py migrate`

### Quick and hacky way

1. In the admin, edit the task, check the "Has unmerged commits" checkbox, and save.
2. Go to that task and click the "Submit Task for Review" button.
3. If you try to actually submit the task, it will fail with an error message.

### Longer and more realistic way

1. Create Dev scratch org.
2. Click "View Org" to open the org.
3. Click the "Object Manager" tab.
4. Click "Contact"
5. Click "Fields & Relationships" on the left.
6. Click the "New" button.
7. Choose "Text" and click the "Next" button.
8. Enter a Field Label (e.g. "Favorite Color") and Length. Click the "Next" button.
9. Deselect all profiles in the Visible column except for System Administrator (not required, but it'll make for less noise in the changes). Click the "Next" button.
10. Deselect all page layouts except for "Contact Layout" (not required). Click the "Save" button.
11. Return to MetaShare and click "check again". It should show 3 uncaptured changes.
12. Click the Capture Task Changes button.
13. Select the field and layout but not the profile. Add a commit message and click the button to capture.
14. The Dev card should show the new commit and one uncaptured change (the profile), and also the "Submit Task for Review" button.
15. Click "Submit Task for Review".
16. You should now be able to submit the form successfully.

### CODE REVIEW

**Comment by jgerigmeyer** [comment]
[I want to submit a task as "ready for review".](https://trello.com/c/s71Jlbye/53-i-want-to-submit-a-task-as-ready-for-review)

File: metashare/api/jobs.py [hunk 1]
```diff
@@ -179,6 +179,10 @@ def commit_changes_from_org(scratch_org, user, desired_changes, commit_message):
         repository = get_repo_info(user, repo_id=repo_id)
         commit = repository.branch(branch).commit
 
+        scratch_org.task.refresh_from_db()
+        scratch_org.task.has_unmerged_commits = True
+        scratch_org.task.save()
```
Comment (jgerigmeyer) [inline metashare/api/jobs.py]: This is a fine first iteration, but I think we'll want this to call a method on the `task` so we can also trigger a WS event.

Comment (wlonk) [reply metashare/api/jobs.py]: Yeah, I wanted to talk with you about that. Do we currently subscribe to Tasks? What info do you need updated on them? When?

Comment (jgerigmeyer) [reply metashare/api/jobs.py]: @wlonk I already made the changes there that I think are helpful. Let me know what you think.

**Comment by jgerigmeyer** [comment]
@wlonk I need to backfill JS test coverage, but this is ready for functional and code review. See also Python changes in https://github.com/oddbird/MetaShare/pull/24/commits/2b3bbf8eded0d7d84174b687a2199ca46c809ffe and https://github.com/oddbird/MetaShare/pull/24/commits/131442a5aa48e5d1350280ff8731b71a32e34c74

**Comment by jgerigmeyer** [comment]
[After submitting a task as ready-for-review, I want to see a link to the GitHub PR on the task page.](https://trello.com/c/xDIqd4Zy/27-after-submitting-a-task-as-ready-for-review-i-want-to-see-a-link-to-the-github-pr-on-the-task-page)

**Comment by jgerigmeyer** [comment]
@stacyk We've decided to stick with the existing "Markdown Guide" for now, but it will likely be replaced by different content in another story. So it's not really worth much time getting the styling perfect. I made some tweaks to shorten the content and give it a bit more space. Feel free to review and make any obvious fixes/cleanup.

**Pull request merged**